### PR TITLE
Add new land realization (transition matrix) and rewrite of the som module 

### PR DESCRIPTION
**Pull request by k4rst3ns**
* include new land realization featuring land transition matrix
* include rewrite of som module featuring update of carbon stocks via som module
* smaller changes in other modules to remove soil carbon related calculations there and bundle them in som
* new settings for ghg_policy (including bugfix of old ssp default setting, which overestimates soil emissions)

### CODE REVIEW

File: config/default.cfg [hunk 1]
```diff
@@ -102,6 +103,7 @@ cfg$gms$c09_gdp_scenario  <- "SSP2"    # def = SSP2
 
 # ***---------------------    10_land   ----------------------------------------
 # * (feb15): default land realization
+# * (landmatrix_dec18): includes a land transition matrix
 cfg$gms$land <- "feb15"          # def = feb15
```
Comment (flohump) [inline config/default.cfg]: Default is still the old realization. Is that correct?

File: modules/59_som/cellpool_aug16/presolve.gms [hunk 2]
```diff
@@ -5,5 +5,5 @@
 *** |  MAgPIE License Exception, version 1.0 (see LICENSE file).
 *** |  Contact: magpie@pik-potsdam.de
 
-p59_carbon_density(t,j,"noncropland")$(sum(noncropland59,pcm_land(j,noncropland59))>0) = p59_som_pool(j,"noncropland")/sum(noncropland59,pcm_land(j,noncropland59));
-p59_carbon_density(t,j,"cropland")$(pcm_land(j,"crop")>0)=  p59_som_pool(j,"cropland") / pcm_land(j,"crop");
+p59_carbon_density(t,j,noncropland59)$(pcm_land(j,noncropland59)>0) = p59_som_pool(j,noncropland59)/pcm_land(j,noncropland59);
```
Comment (flohump) [inline modules/59_som/cellpool_aug16/presolve.gms]: This is only soil carbon, right?

File: modules/80_optimization/nlp_apr17/solve.gms [hunk 3]
```diff
@@ -26,6 +26,7 @@ magpie.holdfixed = 1 ;
 
 $onecho > conopt4.opt
 Tol_Obj_Change = 3.0e-6
+Tol_Feas_Min = 4.0e-8
```
Comment (flohump) [inline modules/80_optimization/nlp_apr17/solve.gms]: Tol_Feas_Min (real): Minimum feasibility tolerance (after scaling). ↵
See Tol_Feas_Max for a discussion of the dynamic feasibility tolerances used by CONOPT.
Range: [3.e-13, 1.e-5]
Default: 4.e-10

File: config/default.cfg (hunk 1 above)
Comment (k4rst3ns) [reply config/default.cfg]: No not correct - new land just be new default.
I have to rerun the test run again!

File: modules/59_som/cellpool_aug16/presolve.gms (hunk 2 above)
Comment (k4rst3ns) [reply modules/59_som/cellpool_aug16/presolve.gms]: yes - which is indicated by the number '59' which stands for som related parameters.

File: modules/80_optimization/nlp_apr17/solve.gms (hunk 3 above)
Comment (k4rst3ns) [reply modules/80_optimization/nlp_apr17/solve.gms]: Just we add this as comment? Or what you suggested with this information?

Comment (flohump) [reply modules/80_optimization/nlp_apr17/solve.gms]: Just as a comment in github. range and default values to put your choice for Tol_Feas_Min into perspective.

**Pull request merged**